### ISSUE TITLE
Raku script checked in?

### ISSUE THREAD

**Issue by davekeeshan** [TUT-ASI/leon3-grlib-gpl-mirror#3]
Hey

I tried to use that raku script available as a file some where that can be checked out?

If i try and copy it from the issue there have been some character replacement going on, and it does not work as a script

    my $url  = ‘https://www.gaisler.com/index.php/downloads/leongrlib’;

should be (fixed quotes):

    my $url  = 'https://www.gaisler.com/index.php/downloads/leongrlib';

and this line:

    my $line = $page.lines.grep(/｢<a href="/products/grlib/grlib-gpl｣/).head;

I don't know what those characters before the <a href is?

To be fair this is the first time I have found raku used any where and I am also struggling to find systems with it installed.

**Comment by davekeeshan** [comment]
Oh there is a new grlib release, hence the interest

**Comment by AlexDaniel** [comment]
@davekeeshan there's no character replacement. In Raku, `‘’`, `“”` and `｢｣` are valid quotes, and these quotes are present in the original script.

> Oh there is a new grlib release, hence the interest

Thanks for letting me know! Is there a subscribe list or something so that I get notified about grlib updates? Otherwise I just have to check manually from time to time…

I updated the repo now.

**Comment by davekeeshan** [comment]
I'll have to relook at raku then as my standard tools and test editors are not working well with that format (CentOS7).

As to subscribe, are you a member of this group:

https://discourse.grlib.community/latest

I still have to review it from time to time, but they did report a new release a few days ago:

https://discourse.grlib.community/t/grlib-gpl-2022-2-released/260

Thanks for updating the repo, I will try again next time if I see it.

**Comment by AlexDaniel** [comment]
Hey @davekeeshan, @douglaskastle. I wonder, are you still using this repo? 😇 

I have finally added some code for automation. Here it is: https://github.com/TUT-ASI/leon3-grlib-gpl-mirror/tree/automation
The script runs every hour or so. For the next year, this repo will always be up to date. After a year, the github token will expire, but hopefully, I'll get a notification about that ahead of time. 😇

There is a chance that the script will fail for the next release because it runs in a completely new environment, but based on my tests it should not happen. If it does, I'll fix it and from then on it'll be good. 🎉 

Let me know if there's anything else I can help you with!

**Comment by AlexDaniel** [comment]
Oh, and also, I have changed the main branch from `master` to `main`. If you're using this repo as a subrepo, you probably need to switch the subrepo to `main`!

_Edit:_ To clarify, this was done primarily to be consistent with my other new projects and their scripts/configs. It makes my life much easier when everything is the same. Also, not a bad idea to move on with the times 😇

**Comment by davekeeshan** [comment]
@AlexDaniel thats for all that, I do still use this repo work, it is great to be able to track this work, they are pushing the maximum size of the arty100, (which is the main board I use) so I am always curious to see if they can keep adding.

I will adjust my mirroring to match your branch change, no dramas.

I keep any eye out for the next release and see if your script works. I personally just use a crontab task on a cloud computer to keep my mirrored work in sync with the original repos (other forked repos) . I tried to do it in github actions, but it was too fiddly. 

In maybe related news, vivado 2024.1 was released last week, they have their riscv version of microblaze, might be good to put it through its paces, relative to the NOELV.
